### ISSUE TITLE
Code gen is out of sync

### ISSUE THREAD

**Issue by thockin** [kubernetes/kubernetes#65988]
@sttts it looks like something is up on master.  Clean checkout, touch staging/src/k8s.io/api/batch/v2alpha1/, then `make generated_files` -- produces a delta.

```
diff --git a/pkg/apis/batch/v2alpha1/zz_generated.conversion.go b/pkg/apis/batch/v2alpha1/zz_generated.conversion.go
index 3a0fe9cd74..058c6a7093 100644
--- a/pkg/apis/batch/v2alpha1/zz_generated.conversion.go
+++ b/pkg/apis/batch/v2alpha1/zz_generated.conversion.go
@@ -29,7 +29,6 @@ import (
        conversion "k8s.io/apimachinery/pkg/conversion"
        runtime "k8s.io/apimachinery/pkg/runtime"
        batch "k8s.io/kubernetes/pkg/apis/batch"
-       batchv1 "k8s.io/kubernetes/pkg/apis/batch/v1"
        core "k8s.io/kubernetes/pkg/apis/core"
 )
 
@@ -216,7 +215,8 @@ func Convert_batch_JobTemplate_To_v2alpha1_JobTemplate(in *batch.JobTemplate, ou
 
 func autoConvert_v2alpha1_JobTemplateSpec_To_batch_JobTemplateSpec(in *v2alpha1.JobTemplateSpec, out *batch.JobTemplateSpec, s conversion.Scope) error {
        out.ObjectMeta = in.ObjectMeta
-       if err := batchv1.Convert_v1_JobSpec_To_batch_JobSpec(&in.Spec, &out.Spec, s); err != nil {
+       // TODO: Inefficient conversion - can we improve it?
+       if err := s.Convert(&in.Spec, &out.Spec, 0); err != nil {
                return err
        }
        return nil
@@ -229,7 +229,8 @@ func Convert_v2alpha1_JobTemplateSpec_To_batch_JobTemplateSpec(in *v2alpha1.JobT
 
 func autoConvert_batch_JobTemplateSpec_To_v2alpha1_JobTemplateSpec(in *batch.JobTemplateSpec, out *v2alpha1.JobTemplateSpec, s conversion.Scope) error {
        out.ObjectMeta = in.ObjectMeta
-       if err := batchv1.Convert_batch_JobSpec_To_v1_JobSpec(&in.Spec, &out.Spec, s); err != nil {
+       // TODO: Inefficient conversion - can we improve it?
+       if err := s.Convert(&in.Spec, &out.Spec, 0); err != nil {
                return err
        }
        return nil
```

**Comment by thockin** [comment]
```
$ make clean_generated 
$ make generated_files
$ git diff
```
That comes out clean.

```
$ find . -name zz_generated.* | xargs rm
$ make generated_files
$ git diff
````

That comes out clean.

That dir, and who knows how many others, is missing some metadata to allow conversions

**Comment by liggitt** [comment]
I'm not seeing that...

```
$ git checkout master

$ git show HEAD
commit f70410959d2801320b528e287f2fcf764f261335 (HEAD -> master, upstream/master, upstream/HEAD, origin/master)

$ make clean
$ make generated_files

$ git status
On branch master
Your branch is up-to-date with 'origin/master'.

nothing to commit, working tree clean
```

do you have a stale `conversion-gen`?

**Comment by liggitt** [comment]
ah, missed the `touch staging/src/k8s.io/api/batch/v2alpha1` step

**Comment by liggitt** [comment]
looks like it might be the v2alpha1 conversion referencing a v1 type... if the generation processes both the v1 and v2alpha1 packages, it finds the v1 conversion. if it only processes the v2alpha1 package, it doesn't and reverts to the reflection-driven path

**Comment by thockin** [comment]
touch staging/src/k8s.io/api/batch/v2alpha1/, then make generated_files

**Comment by thockin** [comment]
try just touching the one

**Comment by sttts** [comment]
```
$ touch pkg/apis/batch/v2alpha1
$ make generated_files
...
./hack/run-in-gopath.sh _output/bin/conversion-gen --extra-peer-dirs k8s.io/kubernetes/pkg/apis/core,k8s.io/kubernetes/pkg/apis/core/v1,k8s.io/api/core/v1 --v 1 --logtostderr -i k8s.io/kubernetes/pkg/apis/batch/v2alpha1 -O zz_generated.conversion
```

That are not many input dirs. The generator does not see any custom conversion funcs in v1.

**Comment by liggitt** [comment]
/assign

fixed in #66030
